### ISSUE TITLE
Hypnogram.hypno as a pd.Categorical?

### ISSUE THREAD

**Issue by raphaelvallat** [raphaelvallat/yasa#125]
@remrama , related to our discussion about the output of `yasa.Hypnogram.upsample_to_data` in #124, I was thinking that we could in fact set the default dtype of `hyp.hypno` as a [pandas.Categorical](https://pandas.pydata.org/pandas-docs/version/1.4/reference/api/pandas.Categorical.html). This reduces the memory size of an upsampled hypnogram (5 stages, 8 hour TIB, freq="10ms") from 22 MB (dtype=str or dtype=np.int64) to 2.7 MB.

```python
pd.Categorical(hyp.hypno, categories=hyp.labels, ordered=False)
```

**Comment by remrama** [comment]
O great idea!! I didn't realize that drops memory so much. Seems like a "go" at first thought, though there might be some subtle and unforeseen limitations. I can't remember any specific examples, but this has been somewhat of my prior experience with `pandas`.Categorical`. I think it's definitely worth trying.

**Comment by raphaelvallat** [comment]
Great! I also had some prior issues, especially when using groupby functions. But I don't think it applies to us so let's make the switch! I can create a separate PR once we've merged #124
